app/models/event.rb

```diff
@@ -177,6 +177,81 @@ def email_addresses_of_type_without_notification_sent(type)
     applications.collect { |a| a.user.email }
   end
 
+  # Returns a string of all email addresses of accepted applications
+  #
+  # @param none
+  # @return [String] Concatenation of all email addresses of accepted applications, seperated by ','
+  def email_addresses_of_accepted_applicants
+    applications = application_letters.where(status: ApplicationLetter.statuses[:accepted])
+    applications.collect { |a| a.user.email }.join(',')
+  end
+
+  # Returns a new email to a set of participants
+  #
+  # @param criteria String
+  # @param value String
+  # @param sender String
+  # @return [Email] new email
+  def generate_participants_email(all, groups, users, sender)
+    email = Email.new
+    email.hide_recipients = false
+    email.recipients = email_addresses_of_participants(all, groups, users)
+    email.reply_to = sender
+    email.subject = ''
+    email.content = ''
+    email
+  end
+
+  # Returns all email addresses of user that fit the given criteria
+  #
+  # @param all Boolean
+  # @param groups [String]
+  # @param users [Int]
+  # @return [String] list of email addresses
+  def email_addresses_of_participants(all, groups, users)
+    if all
+      email_addresses_of_accepted_applicants
+    else
+      (email_addresses_of_groups(groups) + email_addresses_of_users(users)).uniq.join(',')
+    end
+  end
+
+  # Returns a list of tuples containing all participant names and their id
+  #
+  # @return [[String, Int]]
+  def participants_with_id
+    participants.map{|participant| [participant.profile.first_name + ' ' + participant.profile.last_name, participant.id]}
+  end
+
+  # Returns a list of group names and their id
+  #
+  # @return []
+  def groups_with_id
+    existing = ParticipantGroup::GROUPS.select do |group_id,colorcode|
+      participant_groups.where(:group => group_id).count > 0
+    end
+    existing.map do |group_id,colorcode|
+      [I18n.t("participant_groups.options.#{colorcode}"), group_id]
+    end
+  end
+
+  # Returns a list of email addresses of all participants that are in one of the given groups
+  #
+  # @return [String]
+  def email_addresses_of_groups(groups)
+    groups.reduce([]) {|addrs, group| addrs + email_addresses_of_group(group)}.uniq
```

Comment:
>addrs
You're not saving that many letters, you know. Might as well write it out :grin: